packages/core/src/contrib/Constraints.ts

```diff
@@ -541,64 +628,139 @@ const pointInBox = (p: any, rect: any): boolean => {
  * https://stackoverflow.com/questions/5254838/calculating-distance-between-a-point-and-a-rectangular-box-nearest-point
  */
 const dsqBP = (p: any, rect: any): VarAD => {
-  const dx = max(max(sub(rect.minX, p.x), constOf(0.0)), sub(p.x, rect.maxX));
-  const dy = max(max(sub(rect.minY, p.y), constOf(0.0)), sub(p.y, rect.maxY));
-  return add(squared(dx), squared(dy));
+    const dx = max(max(sub(rect.minX, p.x), constOf(0.0)), sub(p.x, rect.maxX));
+    const dy = max(max(sub(rect.minY, p.y), constOf(0.0)), sub(p.y, rect.maxY));
+    return add(squared(dx), squared(dy));
 };
 
 /**
  * Linearly interpolate between left `l` and right `r` endpoints, at fraction `k` of interpolation.
  */
 const lerp = (l: VarAD, r: VarAD, k: VarAD): VarAD => {
-  // TODO: Rewrite the lerp code to be more concise
-  return add(mul(l, sub(constOf(1.0), k)), mul(r, k));
+    // TODO: Rewrite the lerp code to be more concise
+    return add(mul(l, sub(constOf(1.0), k)), mul(r, k));
 };
 
 /**
  * Linearly interpolate between vector `l` and vector `r` endpoints, at fraction `k` of interpolation.
  */
 const lerp2 = (l: VarAD[], r: VarAD[], k: VarAD): [VarAD, VarAD] => {
-  return [lerp(l[0], r[0], k), lerp(l[1], r[1], k)];
+    return [lerp(l[0], r[0], k), lerp(l[1], r[1], k)];
 };
 
 /**
  * Sample a line `line` at `NUM_SAMPLES` points uniformly.
  */
 const sampleSeg = (line: VarAD[][]) => {
-  const NUM_SAMPLES = 15;
-  const NUM_SAMPLES2 = constOf(1 + NUM_SAMPLES);
-  // TODO: Check that this covers the whole line, i.e. no off-by-one error
-  const samples = _.range(1 + NUM_SAMPLES).map((i) => {
-    const k = div(constOf(i), NUM_SAMPLES2);
-    return lerp2(line[0], line[1], k);
-  });
-
-  return samples;
+    const NUM_SAMPLES = 15;
```

Comment:
Strange spacing. Can you check if you are running Prettier in the pre-commit hook?

Reply:
I thought I was. Will check out later tonight...

Reply:
How should I be checking if I'm running Prettier? Maybe we can discuss in the team meeting

Reply:
You should get messages like below when the commit hook is running. We can double-check later.
![image](https://user-images.githubusercontent.com/11740102/109855584-747eca80-7c26-11eb-913b-c5223b7e8669.png)

Reply:
Thanks. Yeah, I'm getting those messages on all my commits (i.e. the hook is running). Not sure why the indentation would be different here given that.